crates/worker/src/partition_processor_manager.rs

```diff
@@ -743,6 +749,45 @@ impl<T: TransportConnect> PartitionProcessorManager<T> {
         }
     }
 
+    fn request_partition_snapshots(&mut self) {
+        let Some(records_per_snapshot) = self
+            .updateable_config
+            .live_load()
+            .worker
+            .snapshots
+            .records_per_snapshot
+        else {
+            return;
+        };
+
+        for (partition_id, status) in self.running_partition_processors.iter() {
+            let ProcessorStatus::Started(state) = status else {
+                continue;
+            };
+            let status = state.watch_rx.borrow().clone();
+            if status.is_effective_leader()
+                && status.replay_status == ReplayStatus::Active
+                && status.last_applied_log_lsn.unwrap_or(Lsn::INVALID)
+                    >= status
+                        .last_archived_log_lsn
+                        .unwrap_or(Lsn::OLDEST)
+                        .add(Lsn::from(records_per_snapshot))
+            {
+                debug!(%partition_id, "Triggering snapshot on partition leader after records_per_snapshot exceeded (last snapshot: {}, current LSN: {})", status.last_archived_log_lsn.unwrap_or(Lsn::OLDEST), status.last_applied_log_lsn.unwrap_or(Lsn::INVALID));
+                let (tx, _) = oneshot::channel();
```

Comment:
I guess we are not interested in the result of the `CreateSnapshot` call here?

Reply:
No - and I possibly should have left a comment explaining why :-) (which is essentially that we can't do much other than log it, and that already happens in the `SnapshotProducer`)